src/Core/src/Stream/StreamFactory.php

```diff
@@ -17,7 +17,7 @@ public static function create($content): Stream
             return StringStream::create($content ?? '');
         }
         if (\is_callable($content)) {
-            return CallableStream::create($content);
+            return RewindableStream::create(CallableStream::create($content));
```

Comment:
Why are we always writing a callable to disk?
Will we always read it twice or is that only a special case?

Reply:
Sadly yes, to compute hash in signature, we have to read the stream first THEN to read it a second time when sending the request.

Note: previous implementation ends to stringify the requests which were worse IMO.

BUT, I've updated the implementaiton to always use memory AND fallback to disk when the buffer exceed 1Mo. That should fixes your concerns

Reply:
Perfect!